### PR TITLE
feat(textarea): add description to TextArea component

### PR DESCRIPTION
**Pull request by omgrief**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @omgrief has signed the CLA. Thanks for the pull request!

**Comment by mickr** [comment]
Please update the title to match the first commit message

**Comment by mickr** [comment]
@omgrief I would make the title `feat(textarea): add description to textarea component` which is similar to the other pull requests to the repo. This will help alleviate confusion if you have to merge manually through Github

File: src/components/text-area/TextArea.js [hunk 1]
```diff
@@ -40,18 +44,33 @@ const TextArea = ({
...
     return (
         <div className={classes}>
-            <Label hideLabel={hideLabel} showOptionalText={!hideOptionalLabel && !isRequired} text={label}>
+            <Label
+                hideLabel={hideLabel}
+                showOptionalText={!hideOptionalLabel && !isRequired}
+                text={label}
+                htmlFor={textAreaID}
+            >
+                {!!description && (
```
Comment (mxiao6) [inline src/components/text-area/TextArea.js]: Is `!!` necessary here?

File: src/components/text-input/TextInput.js [hunk 2]
```diff
@@ -69,10 +75,15 @@ const TextInput = ({
                 showOptionalText={!hideOptionalLabel && !isRequired}
                 text={label}
                 tooltip={labelTooltip}
+                htmlFor={textInputID}
             >
-                {!!description && <i className="text-input-description">{description}</i>}
+                {!!description && (
```
Comment (mxiao6) [inline src/components/text-input/TextInput.js]: Same here

File: src/components/text-input/TextInput.js [hunk 3]
```diff
@@ -69,10 +75,15 @@ const TextInput = ({
                 showOptionalText={!hideOptionalLabel && !isRequired}
                 text={label}
                 tooltip={labelTooltip}
+                htmlFor={textInputID}
```
Comment (jeeyi) [inline src/components/text-input/TextInput.js]: I don't think it's a good idea to do `<label for="something">...</label>` when you use label element as container. It's like trying to do implicit and explicit label association at the same time. Could you remove it?

File: src/components/text-area/TextArea.js [hunk 4]
```diff
@@ -40,18 +44,33 @@ const TextArea = ({
...
     };
 
     return (
         <div className={classes}>
-            <Label hideLabel={hideLabel} showOptionalText={!hideOptionalLabel && !isRequired} text={label}>
+            <Label
+                hideLabel={hideLabel}
+                showOptionalText={!hideOptionalLabel && !isRequired}
+                text={label}
+                htmlFor={textAreaID}
```
Comment (jeeyi) [inline src/components/text-area/TextArea.js]: `htmlFor` can be removed if we keep nesting the form.

Comment (omgrief) [reply src/components/text-area/TextArea.js]: 👍

File: src/components/text-area/TextArea.js (hunk 1 above)
Comment (omgrief) [reply src/components/text-area/TextArea.js]: was just keeping with consistency of `TextInput` component

File: src/components/text-area/__tests__/TextArea.test.js [hunk 5]
```diff
@@ -78,4 +78,10 @@ describe('components/text-area/TextArea', () => {
         const wrapper = shallow(<TextArea label="label" />);
         expect(wrapper).toMatchSnapshot();
     });
+
+    test('should render text area with description', () => {
+        const wrapper = shallow(<TextArea label="label" description="some description" />);
+
+        expect(wrapper).toMatchSnapshot();
```
Comment (jstoffan) [inline src/components/text-area/__tests__/TextArea.test.js]: Ideally, tests should also have explicit expectations in addition to snapshots.

**Comment by jstoffan** [comment]
I spoke offline with @omgrief and I'm going to merge on behalf, as his team hasn't yet been added to the necessary org in Github.

**Pull request merged**